Jint/Native/JsValue.cs

```diff
@@ -385,6 +390,18 @@ public bool Equals(JsValue other)
 
         public Types Type => _type;
 
+        internal static JsValue FromDouble(double value)
+        {
+            // we can cache positive double zero, but not negative, -0 == 0 in C# but in JS it's a different story
+            if ((value == 0 && BitConverter.DoubleToInt64Bits(value) != NegativeZeroBits || value >= 1)
```

Comment:
bit level check for negative made it possible to use cached positive zero, c# evaluates -0 == 0 so that was reason I earlier skipped the caching, but this seems to work as expected